src/ripple/rpc/tests/AccountInfo_test.cpp

```diff
@@ -0,0 +1,162 @@
+            expect (signerLists.isArray());
+            expect (signerLists.size() == 0);
+        }
+
+        // Give alice a SignerList.
+        Account const bogie {"bogie"};
+
+        Json::Value const smallSigners = signers(alice, 2, { { bogie, 3 } });
+        env(smallSigners);
+        {
+            // account_info without the "signer_lists" argument.
+            auto const info = env.rpc ("json", "account_info", withoutSigners);
+            expect (! info[jss::result][jss::account_data].
+                isMember (jss::signer_lists));
+        }
+        {
+            // account_info with the "signer_lists" argument.
+            auto const info = env.rpc ("json", "account_info", withSigners);
+            auto const& data = info[jss::result][jss::account_data];
+            expect (data.isMember (jss::signer_lists));
+            auto const& signerLists = data[jss::signer_lists];
+            expect (signerLists.isArray());
+            expect (signerLists.size() == 1);
+            auto const& signers = signerLists[0u];
+            expect (signers.isObject());
+            expect (signers[sfSignerQuorum.jsonName] == 2);
+            auto const& signerEntries = signers[sfSignerEntries.jsonName];
+            expect (signerEntries.size() == 1);
+            auto const& entry0 = signerEntries[0u][sfSignerEntry.jsonName];
+            expect (entry0[sfSignerWeight.jsonName] == 3);
+        }
+
+        // Give alice a big signer list
+        Account const demon {"demon"};
+        Account const ghost {"ghost"};
+        Account const haunt {"haunt"};
+        Account const jinni {"jinni"};
+        Account const phase {"phase"};
+        Account const shade {"shade"};
+        Account const spook {"spook"};
+
+        Json::Value const bigSigners = signers(alice, 4, {
+            {bogie, 1}, {demon, 1}, {ghost, 1}, {haunt, 1},
+            {jinni, 1}, {phase, 1}, {shade, 1}, {spook, 1}, });
+        env(bigSigners);
+        {
+            // account_info with the "signer_lists" argument.
+            auto const info = env.rpc ("json", "account_info", withSigners);
+            auto const& data = info[jss::result][jss::account_data];
+            expect (data.isMember (jss::signer_lists));
+            auto const& signerLists = data[jss::signer_lists];
+            expect (signerLists.isArray());
+            expect (signerLists.size() == 1);
+            auto const& signers = signerLists[0u];
+            expect (signers.isObject());
+            expect (signers[sfSignerQuorum.jsonName] == 4);
+            auto const& signerEntries = signers[sfSignerEntries.jsonName];
+            expect (signerEntries.size() == 8);
+            auto const& entry0 = signerEntries[0u][sfSignerEntry.jsonName];
+            expect (entry0[sfSignerWeight.jsonName] == 1);
+            auto const& entry7 = signerEntries[7u][sfSignerEntry.jsonName];
+            expect (entry7[sfSignerWeight.jsonName] == 1);
```

Comment:
Yup.  I figured examining the two ends of the array would be sufficient.  But @ximinez's suggestion of a `for` loop is a good one.